src/cmd-cloud-prune

```diff
@@ -414,5 +437,71 @@ def prune_build(s3_client, bucket, prefix, build_id, dry_run):
                 raise Exception(f"Error pruning {build_id}: {e.response['Error']['Message']}")
 
 
+def get_container_tags(meta_json, stream):
+    container_tags = []
+    base_oscontainer = meta_json.get("base-oscontainer")
+    if base_oscontainer:
+        tags = base_oscontainer.get("tags", [])
+        # Only include tags that do not match the stream i.e. moving tags
+        filtered_tags = [tag for tag in tags if tag != stream]
+        if filtered_tags:
+            container_tags = filtered_tags
+    return container_tags
+
+
+def prune_containers(containers_config):
+    barrier_releases = set()
+    # Get the update graph for stable streams
+    if containers_config["stream"] in ['stable', 'testing', 'next']:
+        update_graph = get_update_graph(containers_config["stream"])['releases']
+        # Keep only the barrier releases
+        barrier_releases = set([release["version"] for release in update_graph if "barrier" in release])
+
+    for tag in containers_config["container_tags"]:
+        if tag in STREAMS:
+            continue
+        if tag in barrier_releases:
+            print(f"Release {tag} is a barrier release, keeping.")
+            continue
+        if containers_config["dry_run"]:
+            print(f"Would prune image {containers_config["repository_url"]}:{tag}")
+        else:
+            skopeo_delete(containers_config["repository_url"], tag, containers_config["registry_auth_file"])
+
+
+def get_update_graph(stream):
+    url = f"https://builds.coreos.fedoraproject.org/updates/{stream}.json"
+    r = requests.get(url, timeout=5)
+    if r.status_code != 200:
+        raise Exception(f"Could not download update graph for {stream}. HTTP {r.status_code}")
+    return r.json()
+
+
+def skopeo_inspect(repo, image, auth):
+    skopeo_args = ["skopeo", "inspect", f"docker://{repo}:{image}"]
```

Comment:
we should add [`--no-tags`](https://github.com/containers/skopeo/commit/c2732cb15d3f9e9aab3c4a7db8cb5153e3511a78) here in order to not query for all the tags in the repo:

```suggestion
    skopeo_args = ["skopeo", "inspect", "--no-tags", f"docker://{repo}:{image}"]
```

Reply:
might want to add `--retry-times=10` here too to try to prevent a network flake from tanking us